providers/ethereum-provider/src/EthereumProvider.ts

```diff
@@ -121,18 +127,13 @@ export function buildNamespaces(params: NamespacesParams): {
     throw new Error("Invalid chains");
   }
 
-  const requiredChains = chains;
-  const requiredMethods = methods || REQUIRED_METHODS;
-  const requiredEvents = events || REQUIRED_EVENTS;
-  const requiredRpcMap = {
-    [getEthereumChainId(requiredChains)]: rpcMap[getEthereumChainId(requiredChains)],
-  };
-
   const required: Namespace = {
-    chains: requiredChains,
-    methods: requiredMethods,
-    events: requiredEvents,
-    rpcMap: requiredRpcMap,
+    chains,
+    methods: methods || REQUIRED_METHODS,
+    events: events || REQUIRED_EVENTS,
+    rpcMap: {
+      ...(chains.length && { [getEthereumChainId(chains)]: rpcMap[getEthereumChainId(chains)] }),
```

Comment:
Seems like this may attempt to spread `0` if `chains.length === 0`. Maybe safer with a ternary and always returning an object?

```suggestion
      ...(chains.length ? { [getEthereumChainId(chains)]: rpcMap[getEthereumChainId(chains)] } : {}),
```